Geometry/TrackerCommonData/plugins/dd4hep/DDTECModuleAlgo.cc

```diff
@@ -409,6 +409,7 @@ static long algorithm(Detector& /* description */, cms::DDParsingContext& ctxt,
     dy = 0.5 * pitchThick;
     dz = 0.5 * pitchHeight;
     solid = Box(dx, dy, dz);
+    name += "Box";
```

Comment:
@cvuosalo - why? It’s a `Box` and you can always check it by type.

Reply:
This problem relates to namespaces. In the old DD, all component names are qualified by namespaces, so these definitions result:
```
<Box name="tecmodule0r:TECModule0PA" dx="43.9000*mm" dy="0.3325*mm" dz="6.5000*mm"/>
<Trapezoid name="tecmodule0s:TECModule0PA" dz="43.9000*mm" theta="2.8576*deg" phi="0.0000*deg" h1="0.3325*mm" bl1="9.3663*mm" tl1="9.3663*mm" alp1="0.0000*deg" h2="0.3325*mm" bl2="4.9837*mm" tl2="4.9837*mm" alp2="0.0000*deg"/>
```
but with DD4hep, there is no namespace for these names, so WITHOUT this fix these solid definitions result :
```
<Box name="TECModule0PA" dx="43.9000*mm" dy="0.3325*mm" dz="6.5000*mm"/>
<Trapezoid name="TECModule0PA" dz="43.9000*mm" theta="2.8576*deg" phi="0.0000*deg" h1="0.3325*mm" bl1="9.3663*mm" tl1="9.3663*mm" alp1="0.0000*deg" h2="0.3325*mm" bl2="4.9837*mm" tl2="4.9837*mm" alp2="0.0000*deg"/>
```
Here are two different solids with the same name. I don't see how that could work correctly. Running the code with the appended names gives:
```
<Box name="TECModule0PABox" dx="43.9000*mm" dy="0.3325*mm" dz="6.5000*mm"/>
<Trapezoid name="TECModule0PATrap" dz="43.9000*mm" theta="2.8576*deg" phi="0.0000*deg" h1="0.3325*mm" bl1="9.3663*mm" tl1="9.3663*mm" alp1="0.0000*deg" h2="0.3325*mm" bl2="4.9837*mm" tl2="4.9837*mm" alp2="0.0000*deg"/>
```

Reply:
Is it not accidental that one is a Box and the other a Trapezoid?
BTW: what the box is used for in reconstruction?

Reply:
I don't know why these two solids were defined as a Box and a Trapezoid. I can only say that this feature is in the old DD and has been there for a long time.